recipes-soletta/dev-app/soletta-dev-app_0.0.7.bb

```diff
@@ -37,62 +33,57 @@ FILES_${PN} += " \
 
 SYSTEMD_SERVICE_${PN} = "soletta-dev-app-server.service soletta-dev-app-avahi-discover.service"
 
-do_compile() {
-
-    # changing the home directory to the working directory, the .npmrc will be created in this directory
-    export HOME=${WORKDIR}
-
-    # does not build dev packages
-    npm config set dev false
-
-    # access npm registry using http
-    npm set strict-ssl false
-    npm config set registry http://registry.npmjs.org/
-
-    # configure http proxy if neccessary
-    if [ -n "${http_proxy}" ]; then
-        npm config set proxy ${http_proxy}
-        export bower_https_proxy=${http_proxy}
-    fi
-    if [ -n "${HTTP_PROXY}" ]; then
-        npm config set proxy ${HTTP_PROXY}
-        export bower_https_proxy=${HTTP_PROXY}
-    fi
-
-    # configure cache to be in working directory
-    npm set cache ${WORKDIR}/npm_cache
-
-    # clear local cache prior to each compile
-    npm cache clear
-
-    case ${TARGET_ARCH} in
-        i?86) targetArch="ia32"
-            echo "targetArch = 32"
-            ;;
-        x86_64) targetArch="x64"
-            echo "targetArch = 64"
-            ;;
-        arm) targetArch="arm"
-            ;;
-        mips) targetArch="mips"
-            ;;
-        sparc) targetArch="sparc"
-            ;;
-        *) echo "unknown architecture"
-           exit 1
-            ;;
-    esac
-
-    npm install --arch=${targetArch} --production --verbose -g bower@v1.7.9
-
-    # compile and install node modules in source directory
-    npm --arch=${targetArch} --production --verbose install
-
-    bower -V install
+do_configure_append() {
+  export HOME=${WORKDIR}
+
+  npm config set dev false
+  npm set strict-ssl false
+  npm config set registry http://registry.npmjs.org/
+
+  if [ -n "${http_proxy}" ]; then
+      npm config set proxy ${http_proxy}
+  fi
+  if [ -n "${HTTP_PROXY}" ]; then
+      npm config set proxy ${HTTP_PROXY}
+  fi
+
+  case ${TARGET_ARCH} in
+      i?86) targetArch="ia32"
+          echo "targetArch = 32"
+          ;;
+      x86_64) targetArch="x64"
+          echo "targetArch = 64"
+          ;;
+      arm) targetArch="arm"
+          ;;
+      mips) targetArch="mips"
+          ;;
+      sparc) targetArch="sparc"
+          ;;
+      *) echo "unknown architecture"
+         exit 1
+          ;;
+  esac
+
+  #Installing bower a client-side libs manager
+  npm install --arch=${targetArch} --production --verbose -g bower@v1.7.9
+}
+
+do_compile_append() {
+
+  if [ -n "${http_proxy}" ]; then
+      export bower_https_proxy=${http_proxy}
+  fi
+  if [ -n "${HTTP_PROXY}" ]; then
+      export bower_https_proxy=${HTTP_PROXY}
+  fi
+
+  #Installing client-side libs
```

Comment:
Ok, so if there are no compiled dependencies for soletta-dev-app, then just distributing it all in a tarball makes a lot of sense.

From: Bruno Bottazzini [mailto:notifications@github.com]
Sent: Monday, April 25, 2016 5:05 PM
To: solettaproject/meta-soletta <meta-soletta@noreply.github.com>
Cc: Puustinen, Ismo <ismo.puustinen@intel.com>; Mention <mention@noreply.github.com>
Subject: Re: [solettaproject/meta-soletta] Replace npm installation to use the fetcher from Yocto (#72)


In recipes-soletta/dev-app/soletta-dev-app_0.0.7.bb<https://github.com/solettaproject/meta-soletta/pull/72#discussion_r60917917>:

> +  esac

> +

> +  #Installing bower a client-side libs manager

> +  npm install --arch=${targetArch} --production --verbose -g bower@v1.7.9<mailto:bower@v1.7.9>

> +}

> +

> +do_compile_append() {

> +

> +  if [ -n "${http_proxy}" ]; then

> +      export bower_https_proxy=${http_proxy}

> +  fi

> +  if [ -n "${HTTP_PROXY}" ]; then

> +      export bower_https_proxy=${HTTP_PROXY}

> +  fi

> +

> +  #Installing client-side libs

@ipuustin<https://github.com/ipuustin>, The only thing that is compiled is the Selenium WebDriver (Server side) to run npm test and this web-drivers should not be distributed at all. There was an error in the standalone package that I fixed it. You can find more info about these web-drivers here: http://angular.github.io/protractor/#/browser-setup

In anycase I can distribute this semi-standalone with only client-libs and we would avoid this problem with bower, although I still prefer using the same package clearlinux uses.

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/solettaproject/meta-soletta/pull/72/files/51e0d286cd094dd0ac1d10640735cefcb068f610#r60917917>